### PR TITLE
try using narwhals to make dataframe-agnostic

### LINKED ISSUE #35: Experiment with `narwhals`
# Description

[narwhals](https://github.com/MarcoGorelli/narwhals) is quickly developing, but it could become a solid building-block/layer of abstraction.

Let's experiment with this!

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #35 

was curious to try this out - curious to hear your results (no offense / worries whatsoever if you don't want or like this!), tbh I think it looks quite neat

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -37,6 +37,7 @@ issue-tracker = "https://github.com/fbruzzesi/timebasedcv/issues"
 
 [project.optional-dependencies]
 polars = ["polars"]
+pandas = ["pandas"]
```
Comment (FBruzzesi) [inline pyproject.toml:40]: Are there min versions required by narwhals?

**Review by FBruzzesi — APPROVED** [review]
Grazie Marco! This looks neat 👌
Will update docs and make a release soon ✨

File: pyproject.toml (hunk 1 above)
Comment (MarcoGorelli) [reply pyproject.toml:40]: thanks!

I'll test this backwards and see how far we can get, but `.filter` I'd expect should go back essentially forever

Comment (MarcoGorelli) [reply pyproject.toml]: would pandas>=1.2.0 and Polars>=0.20.3 work for you? I've set those as the minimum, and have tested with the minimum versions

Comment (FBruzzesi) [reply pyproject.toml]: LGTM

**Review by FBruzzesi — APPROVED** [review]

**Pull request merged**